fv3core/decorators.py

```diff
@@ -169,6 +169,40 @@ def _mark_cuda_fields_written(self, fields: Mapping[str, Storage]):
                 fields[write_field]._set_device_modified()
 
 
+def get_stencils_with_varied_bounds(
+    func: Callable[..., None],
+    origins: List[Index3D],
+    domains: List[Index3D],
+    stencil_config: Optional[StencilConfig] = None,
+    externals: Optional[Mapping[str, Any]] = None,
+):
+    assert len(origins) == len(domains), (
+        "Lists of origins and domains need to have the same length, you provided "
+        + str(len(origins))
+        + " origins and "
+        + str(len(domains))
+        + " domains"
+    )
+    if externals is None:
+        externals = {}
+    if stencil_config is None:
+        stencil_config = global_config.get_stencil_config()
+    stencils = []
+    for origin, domain in zip(origins, domains):
+        ax_offsets = fv3core.utils.grid.axis_offsets(spec.grid, origin, domain)
+        externals = {**externals, **ax_offsets}
```

Comment:
This might mask the `externals` defined in the parent scope (outside the loop), suggested:
```python
    ...
    stencil_config=stencil_config,
    externals={**externals, **ax_offsets},
)
```

Reply:
good catch! thank you